### PR TITLE
feat: add possibility to pass application_name via postgres specific config

### PR DESCRIPTION
**Pull request by das-mensch**
### Description of change

As the "extra" parameter should be deprecated in the future, the need for
passing driver specific options via the respective config objects is given.
The addition of the "applicationName" parameter is a step forward in this 
direction. This parameter is visible in logs and statistics and helps db
administrators to see which connection belongs to which application.
This change also enables the user to use code completion in case they
didn't even knew this options exists

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run lint` passes with this change
- [x] `npm run test` passes with this change
- [ ] This pull request links relevant issues as `Fixes #0000` N/A
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/driver/postgres/PostgresDriver.ts [hunk 1]
```diff
@@ -1101,7 +1101,8 @@ export class PostgresDriver implements Driver {
             database: credentials.database,
             port: credentials.port,
             ssl: credentials.ssl,
-            connectionTimeoutMillis: options.connectTimeoutMS
+            connectionTimeoutMillis: options.connectTimeoutMS,
+            application_name: options.applicationName || "typeorm app"
```
Comment (imnotjames) [inline src/driver/postgres/PostgresDriver.ts:1105]: Can this default to undefined instead

**Review by imnotjames — CHANGES_REQUESTED** [review]

File: src/driver/postgres/PostgresDriver.ts (hunk 1 above)
Comment (das-mensch) [reply src/driver/postgres/PostgresDriver.ts:1105]: sure, my thought was: if any typeorm app is visible with "typeorm app" it'll help to distinguish between typeorm apps and all other apps in the database overview. A bit more "clean by default". But i see your point...

File: src/driver/postgres/PostgresConnectionOptions.ts [hunk 2]
```diff
@@ -67,4 +67,10 @@ export interface PostgresConnectionOptions extends BaseConnectionOptions, Postgr
      * Automatically install postgres extensions
      */
      readonly installExtensions?: boolean;
+
+    /**
+     * sets the application_name var to help db administrators identify
+     * the service using this connection. Defaults to 'typeorm app'
```
Comment (shrujalshah28) [inline src/driver/postgres/PostgresConnectionOptions.ts:73]:
```suggestion
     * the service using this connection. Defaults to 'undefined'
```

**Review by shrujalshah28 — CHANGES_REQUESTED** [review]

File: src/driver/postgres/PostgresConnectionOptions.ts (hunk 2 above)
Comment (das-mensch) [reply src/driver/postgres/PostgresConnectionOptions.ts:73]: yeah saw that a bit too late. i've added it while you where reviewing i think, sorry

**Review by imnotjames — APPROVED** [review]

**Review by shrujalshah28 — APPROVED** [review]

**Pull request merged**